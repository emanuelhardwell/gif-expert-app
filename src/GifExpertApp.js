import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    "One Punch",
    "Samurai X",
    "Dragon Ball",
  ]);

  const handleAdd = () => {
    setCategories([...categories, "HunterXHunter"]);
    // forma 2 de hacerlo
    // setCategories((cats) => [...cats, "HunterXHunter"]);
  };

  //   RETURN
  return (
    <div className="m-3">
      <div className="container">
        <div className="row">
          <div className="col-md-12">
            <h1 className="text-center">GifExpertApp</h1>
            <hr />
          </div>
        </div>

        <div className="row">
          <div className="col-md-12">
            <AddCategory />
          </div>
        </div>

        <div className="row">
          <ol>
            {categories.map((category) => (
              <li key={category}> {category} </li>
            ))}
          </ol>
        </div>

        <div className="row">
          <div className="col-md-4">
            <button className="btn btn-primary" onClick={handleAdd}>
              {" "}
              Agregar category{" "}
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};
