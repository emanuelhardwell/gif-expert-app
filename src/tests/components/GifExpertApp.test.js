import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from "../../GifExpertApp";

describe("Pruebas en el componente GifExpertApp.js", () => {
  test("Deberia estar el snapshot cprrectamente", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Deberia estar el snapshot cprrectamente y mostrar el numero de GIFGRID", () => {
    const defaultCategories = ["Goku", "Kakaroto", "Vegeta"];

    const wrapper = shallow(
      <GifExpertApp defaultCategories={defaultCategories} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(defaultCategories.length);
  });
});
