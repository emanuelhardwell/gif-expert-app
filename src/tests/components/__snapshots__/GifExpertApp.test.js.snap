// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp.js Deberia estar el snapshot cprrectamente 1`] = `
<div
  className="m-2"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <h1
          className="text-center"
        >
          GifExpertApp
        </h1>
        <hr />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4"
      >
        <AddCategory
          setCategories={[Function]}
        />
      </div>
    </div>
    <div
      className="row mt-3  bg-primary"
    />
    <div
      className="row"
    >
      <div
        className="col-md-4"
      >
        <button
          className="btn btn-primary"
        >
           Agregar category 
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en el componente GifExpertApp.js Deberia estar el snapshot cprrectamente y mostrar el numero de GIFGRID 1`] = `
<div
  className="m-2"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <h1
          className="text-center"
        >
          GifExpertApp
        </h1>
        <hr />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4"
      >
        <AddCategory
          setCategories={[Function]}
        />
      </div>
    </div>
    <div
      className="row mt-3  bg-primary"
    >
      <GifGrid
        category="Goku"
        key="Goku"
      />
      <GifGrid
        category="Kakaroto"
        key="Kakaroto"
      />
      <GifGrid
        category="Vegeta"
        key="Vegeta"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4"
      >
        <button
          className="btn btn-primary"
        >
           Agregar category 
        </button>
      </div>
    </div>
  </div>
</div>
`;
